Save space in Preferences -> File Browser Tab

TooWaBoo opened the pull request:
Changed `Folder-open.png` to `Folder-open-small.png`. The TWB-theme is using the same height for buttons and entry fields. The big icon is increasing the size. I hope this is OK

Changed "Context Menu Options" to two columns

Changed the `cache button separator` to a real separater with ID `#PrefCacheSeparator` for styling in css.
```
separator#PrefCacheSeparator.horizontal {
    <any style>
}
```
Adapted the TWB

ping @Beep6581 

old
![oldrt](https://user-images.githubusercontent.com/13800920/47979074-6c97d580-e0c1-11e8-85d0-a3716324397d.PNG)
new
![newrt](https://user-images.githubusercontent.com/13800920/47979082-73264d00-e0c1-11e8-822e-3a8b86757562.PNG)
old
![oldtwb](https://user-images.githubusercontent.com/13800920/47979089-791c2e00-e0c1-11e8-8dcf-a0cfd7cbde55.PNG)
new
![newtwb](https://user-images.githubusercontent.com/13800920/47979093-7de0e200-e0c1-11e8-8026-640afd807327.PNG)

Beep6581 commented:
@TooWaBoo looks good, but I'd like to request one change before we merge.

As it's likely we'll need to use grid row/column separators in other places, they should be assigned a class and not an ID:
```
cacheSeparator->get_style_context()->add_class("grid-row-separator");
```
Paving way for `grid-column-separator` if needed.

RawTherapee currently uses IDs incorrectly in other places, but I'll leave those for 5.6.

TooWaBoo commented:
In general your are right.
eg. in html names are used as an unique ID of an object.
Classes are used:
- to group objects for styling
- if you you need more than on class so the same object can be in more than one group
- if a class changes during runtime.

In the case of a `separator` it's not necessary to have a name ( ID ) or a class. You can simply identify the separator by `grid > separator.horizontal` instead of the class `grid-row-separator` or `grid > separator.vertical`  instead of a `grid-column-separator`.

I've created the `#PrefCacheSeparator` ID to make this separater unique, as you wrote: "...so that a button is not accidentally clicked...". This ID allows you to change e.g. the height of this separator without changing the default height.

In the TWB theme I've set this as default style:
```
grid separator.horizontal {
    margin: 0.166666666666666666em 0;
    padding: 0;
}
```

... but for the cache button separator and only for this I want this style:
```
separator#PrefCacheSeparator.horizontal {
    margin: 0.25em 0.166666666666666666em;
    padding: 0;
}
```
This is the reason why I've created an unique ID and not a class.

TooWaBoo commented:
@Beep6581 
... but if you want I can add a `grid-row-separator` class too.

Thanatomanic commented:
@TooWaBoo Can you use [`calc`](https://developer.mozilla.org/en-US/docs/Web/CSS/calc) here to make things a bit cleaner? So, `calc(1/6)` instead of the repeating digits?

Beep6581 commented:
I just want this to be simple and maintainable, and don't currently understand why there is a need to have a unique ID for this separator. Why is it special? It seems that all horizontal separators used in a grid should be identical.

TooWaBoo commented:
@Beep6581 
Because you wrote this comment in the code: "Separation is needed so that a button is not accidentally clicked when one wanted to click a spinbox", I thought you want this huge separater between the `spinbutton` and `'clear'-button.` To give you the possibility to define any hight in this special case I added the ID.  I'm OK with the a default height of the separator.

Removed the `#PrefCacheSeparator` name in the TWB theme.
Replaced the `#PrefCacheSeparator` name with class `grid-row-separator` in `preferences.cc`

@Thanatomanic 
I'm using so many decimals to avoid rounding issues between GTK 3.22 and 3.24. When I use calc GTK is rounding different in combination of padding, margin and min-height.
